### PR TITLE
Add module for `spring/spring-cloud-config`

### PR DESCRIPTION
**Pull request by Sgitario**
Verifies that we can use an external Spring Cloud Server to inject configuration in our Quarkus applications.
Fix https://github.com/quarkus-qe/quarkus-test-suite/issues/184

### CODE REVIEW

**Comment by Sgitario** [comment]
@jsmrcka would you mind taking a look into this PR?

File: spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.Container;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class SpringCloudConfigIT {
+
+    @Container(image = "${spring.cloud.server.image}", port = 8888, expectedLog = "Started ConfigServer")
+    static RestService spring = new RestService()
+            .withProperty("SPRING_PROPERTY_FILE", "resource::/config/application-SpringCloudConfigIT.properties")
```
Comment (jsmrcka) [inline spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java:19]: I was unable to find any mention of `SPRING_PROPERTY_FILE` anywhere. Can you perhaps point me in the right direction?

**Review by jsmrcka — APPROVED** [review]

File: spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java (hunk 1 above)
Comment (Sgitario) [reply spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java:19]: This is used just to map the external file from the classpath to the container

Comment (jsmrcka) [reply spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java:19]: But how is it used by the container?
I mean, it obvoiusly works, I just don't see how.

Comment (Sgitario) [reply spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java:19]: The property is ignored by the container, but the framework will detect that there is a resource that needs to be mapped into the container.

Comment (jsmrcka) [reply spring/spring-cloud-config/src/test/java/io/quarkus/ts/spring/cloud/config/SpringCloudConfigIT.java:19]: I see. I didn't realize it's the external resources feature of the framework. Thanks for the explanation! :)

**Pull request merged**